webapp-django/crashstats/crashstats/static/crashstats/js/socorro/exploitability_report.js

```diff
@@ -0,0 +1,119 @@
+            // you're on the first page
+            $('a.first', parent)
+                .attr('href', '#1')
+                .addClass('impossible');
+        } else {
+            $('a.first', parent)
+                .attr('href', '#1')
+                .removeClass('impossible');
+        }
+
+        // the "Previous" link
+        if (page > 1) {
+            $('a.previous', parent)
+                .attr('href', '#' + (page - 1))
+                .addClass('possible')
+                .removeClass('impossible');
+        } else {
+            $('a.previous', parent)
+                .attr('href', '#1')
+                .addClass('impossible');
+        }
+
+        // the "Next" link
+        if (page * size < count) {
+            $('a.next', parent)
+                .attr('href', '#' + (page + 1))
+                .removeClass('impossible');
+        } else {
+            $('a.next', parent)
+                .attr('href', '#' + page)
+                .addClass('impossible');
+        }
+
+        // The "Last" link
+        if (page * size < count) {
+            $('a.last', parent)
+                .attr('href', '#' + Math.ceil(count / size))
+                .removeClass('impossible');
+        } else {
+            $('a.last', parent)
+                .attr('href', '#' + Math.ceil(count / size))
+                .addClass('impossible');
+        }
+    }
+
+    function fetchReports(page) {
+        var url = document.location.pathname + document.location.search;
+        url += '&format=json';
+
+        return $.getJSON(url, {page: page})
+        .done(function(response) {
+            var tbody = $('table.data-table tbody');
+            $('tr', tbody).remove();
+            $.each(response.crashes, function(i, crash) {
+                var tr = $('<tr>');
+                tr.append(
+                    $('<td>').append(
+                        $('<a>')
+                        .attr('href', crash.url)
+                        .text(crash.signature)
+                    )
+                );
+                var bugsHTML = '';
+                $.each(crash.bugs, function(i, bug) {
+                    bugsHTML += bug.html + ' ';
+                });
+                tr.append(
+                    $('<td>').addClass('bug_ids_more').html(bugsHTML)
+                );
+                tr.append($('<td>').text(crash.high_count));
+                tr.append($('<td>').text(crash.medium_count));
+                tr.append($('<td>').text(crash.low_count));
+                tr.append($('<td>').text(crash.none_count));
+                tbody.append(tr);
+            });
+            $('.pagination').show();
+            updatePaginationLinks(
+                page,
+                response.count,
+                response.batch_size
+            );
+            BugLinks.enhance();
+        })
+        .fail(function() {
+            console.error.apply(console, arguments);
+            alert("Loading reports failed.");
```

Comment:
Mmm, don't you want to show an error in the HTML content instead? I find alerts to be pretty aggressive.